patches/minecraft/net/minecraft/block/AbstractRailBlock.java.patch

```diff
@@ -1,19 +1,24 @@
 --- a/net/minecraft/block/AbstractRailBlock.java
 +++ b/net/minecraft/block/AbstractRailBlock.java
-@@ -13,7 +_,7 @@
+@@ -13,9 +_,10 @@
  import net.minecraft.world.IWorldReader;
  import net.minecraft.world.World;
  
 -public abstract class AbstractRailBlock extends Block {
 +public abstract class AbstractRailBlock extends Block implements net.minecraftforge.common.extensions.IAbstractRailBlock {
     protected static final VoxelShape field_185590_a = Block.func_208617_a(0.0D, 0.0D, 0.0D, 16.0D, 2.0D, 16.0D);
     protected static final VoxelShape field_190959_b = Block.func_208617_a(0.0D, 0.0D, 0.0D, 16.0D, 8.0D, 16.0D);
++   public static net.minecraft.state.EnumProperty<RailShape> SHAPE = net.minecraft.state.properties.BlockStateProperties.field_208165_R;
     private final boolean field_196277_c;
-@@ -37,6 +_,7 @@
+ 
+    public static boolean func_208488_a(World p_208488_0_, BlockPos p_208488_1_) {
+@@ -36,7 +_,8 @@
+    }
  
     public VoxelShape func_220053_a(BlockState p_220053_1_, IBlockReader p_220053_2_, BlockPos p_220053_3_, ISelectionContext p_220053_4_) {
-       RailShape railshape = p_220053_1_.func_203425_a(this) ? p_220053_1_.func_177229_b(this.func_176560_l()) : null;
-+      RailShape railShape2 = p_220053_1_.func_203425_a(this) ? getRailDirection(p_220053_1_, p_220053_2_, p_220053_3_, null) : null;
+-      RailShape railshape = p_220053_1_.func_203425_a(this) ? p_220053_1_.func_177229_b(this.func_176560_l()) : null;
++      //RailShape railshape = p_220053_1_.is(this) ? p_220053_1_.getValue(this.getShapeProperty()) : null;
++      RailShape railshape = p_220053_1_.func_203425_a(this) ? getRailDirection(p_220053_1_, p_220053_2_, p_220053_3_, null) : null;
```

Comment:
This is a strange patch. Why is there a comment here?

Reply:
The existing code had both versions (getShapeProperty and getRailDirection) but assigned from the property.
I assume there was some reason why both function calls are here, so rather than outright deleting the first, I just commented it and changed the assignment to use the existing getRailDirection call.

Reply:
It seems this patch was broken in the 1.16 update. Look at the Git history of the patch file.